src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -1820,73 +1821,82 @@ public virtual IndexProgress GetProgress(DocumentsOperationContext documentsCont
         {
             using (CurrentlyInUse(out var valid))
             {
+                if (documentsContext.Transaction == null)
+                    throw new InvalidOperationException("Cannot calculate index progress without valid transaction.");
+
                 if (valid == false || DocumentDatabase.DatabaseShutdown.IsCancellationRequested || _disposeOne.Disposed)
                 {
-                    return new IndexProgress
+                    var progress = new IndexProgress
                     {
                         Name = Name,
-                        Type = Type
+                        Type = Type,
+                        IndexRunningStatus = Status,
+                        Collections = new Dictionary<string, IndexProgress.CollectionStats>(StringComparer.OrdinalIgnoreCase)
                     };
+
+                    UpdateIndexProgress(documentsContext, progress, null);
```

Comment:
Is this safe? We can have a `DatabaseShutdown` or `_disposeOne.Disposed` here.

Reply:
We check that before that:
https://github.com/ravendb/ravendb/pull/7973/files#diff-a98877097408ea1a038c4101886f9a9bR1827

Reply:
Yes, and then you are executing this method here, even if those are true?

Reply:
Correct! fixed